### PR TITLE
Typescript/audit docs

### PR DESCRIPTION
**Pull request by ambaldwin**
# Submit a pull request

## CLA

- [ ] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required).
- [ ] The code changes follow best practices
- [ ] Code changes are tested (add some information if not applicable)

## Description of the pull request
Audit docs as part of our typescript conversion.
Please put special review on the descriptions in ChannelContext as I had to piece together these.

This PR also excludes any updates to the docs for the MessageInput components  as these are still being typescripted.

### CODE REVIEW

File: src/components/Message/Message.tsx [hunk 1]
```diff
@@ -184,8 +184,22 @@ const UnMemoizedMessage = <
...
+ * - **isAdmin** returns true if current user has admin role.
+ * - **canEditMessage** returns true if current user has permission to edit message.
+ * - **canDeleteMessage** returns true if current user has permission to edit message.
+ * - **handleFlag** Handler to flag a message
+ * - **handleMute** Handler to mute a user of message
+ * - **handleEdit** Handler to edit a message
+ * - **handleDelete** Handler to delete a message
+ * - **handleReaction** Handler to add/remove reaction on message
+ * - **handleRetry** Handler to resend the message, in case of failure.
+ * - **openThread** Handler to open the thread on current message.
```
Comment (DanC5) [inline src/components/Message/Message.tsx:202]: this isn't an exhaustive list and also some of these methods are old, let's delete

File: src/components/Message/types.ts [hunk 2]
```diff
@@ -190,84 +181,84 @@ export type MessageUIComponentProps<
...
   /** Function to exit edit state */
   clearEditingState: (
     event?: React.MouseEvent<HTMLElement, globalThis.MouseEvent> | undefined,
   ) => void;
-  /** If the message is in edit state */
+  /** If the Message is in edit state */
   editing: boolean;
   /**
-   * Returns all allowed actions on message by current user e.g., ['edit', 'delete', 'flag', 'mute', 'react', 'reply']
+   * Returns all allowed actions on message by current user e.g., ['edit', 'delete', 'flag', 'mute', 'react', 'reply'].
```
Comment (DanC5) [inline src/components/Message/types.ts:193]: also everywhere we have an actions array add 'pin' in abc order

File: src/components/MessageActions/MessageActionsBox.tsx [hunk 3]
```diff
@@ -149,6 +149,10 @@ const UnMemoizedMessageActionsBox = <
   );
 };
 
+/**
+ * MessageActionsBox - Allows the user to flag, edit or delete a message. There's also a command(`/mute`) for muting the user that wrote the message.
```
Comment (DanC5) [inline src/components/MessageActions/MessageActionsBox.tsx:153]: this sounds a little funky, let's replace with "Allows a user to edit, (add the rest) a message."

File: src/docs/ChatContext.md [hunk 4]
```diff
@@ -1,10 +1,16 @@
...
+- **openMobileNav** A function to open mobile navigation
 - **setActiveChannel** A function to set the currently active channel. This is used in [ChannelList](#channellist) component to navigate between channels.
 
   **Params**
 
-  - `channel` Channel that needs to set to as active channel.
-
-- **channel** The currently active channel
+  - `newChannel` Channel that needs to be set as active Channel
+  - `watchers` water parameters: { limit?: number; offset?: number }
```
Comment (DanC5) [inline src/docs/ChatContext.md:15]: water?

Comment (ambaldwin) [reply src/docs/ChatContext.md:15]: whoops! Searched the whole project for water to be sure I didn't copy and paste this elsewhere too...

**Review by DanC5 — APPROVED** [review]

**Pull request merged**